lib/kafka/offset_manager.rb

```diff
@@ -40,7 +40,7 @@ def commit_offsets
         @group.commit_offsets(@processed_offsets)
 
         @last_commit = Time.now
-        @processed_offsets.clear
```

Comment:
The idea is that the consumer will simply keep using and updating `@processed_offsets`. Only when it has to rejoin the group does it clear its cache and re-sync.